centreon/src/Utility/EnvironmentFileManager.php

```diff
@@ -0,0 +1,182 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Utility;
+
+class EnvironmentFileManager
+{
+    /** @var array<string, int|float|bool|string> */
+    private array $variables = [];
+
+    private ?string $currentEnvironnementFile = null;
+
+    /**
+     * @param string $environmentFilePath
+     */
+    public function __construct(private string $environmentFilePath)
+    {
+        $this->environmentFilePath = $this->addDirectorySeparatorIfNeeded($this->environmentFilePath);
+    }
+
+    /**
+     * @param string $environmentFile
+     *
+     * @throws \Exception
+     */
+    public function load(string $environmentFile = '.env'): void
+    {
+        $this->currentEnvironnementFile = $environmentFile;
+        $filePath = $this->environmentFilePath . $environmentFile;
+        if (! file_exists($filePath)) {
+            throw new \Exception(sprintf("The environment file '%s' does not exist", $environmentFile));
+        }
+
+        $file = fopen($filePath, 'r');
+        if (! $file) {
+            throw new \Exception(sprintf('Impossible to open file \'%s\'', $filePath));
+        }
+        try {
+            while (($line = fgets($file, 1024)) !== false) {
+                $line = trim($line);
+                if (str_contains($line, '=')) {
+                    [$key, $value] = explode('=', $line);
+                    $this->add($key, $value);
+                }
+            }
+        } finally {
+            fclose($file);
+        }
+    }
+
+    /**
+     * @param string $key
+     * @param int|float|bool|string $value
+     */
+    public function add(string $key, int|float|bool|string $value): void
+    {
+        if ($value === 'true' || $value === 'false') {
+            $value = (bool) $value;
+        } elseif (preg_match('/^-?\d+\.\d+/', (string) $value)) {
+            $value = (float) $value;
+        } elseif (is_numeric($value)) {
+            if (str_starts_with($key, 'IS_') && $value === '1') {
+                $value = (bool) $value;
+            } else {
+                $value = (int) $value;
+            }
+        }
+        $this->variables[$key] = $value;
+    }
+
+    /**
+     * @param string $key
+     */
+    public function delete(string $key): void
+    {
+        unset($this->variables[$key]);
+    }
+
+    /**
+     * @return array<string, int|float|bool|string>
+     */
+    public function getAll(): array
+    {
+        return $this->variables;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return int|float|bool|string|null
+     */
+    public function get(string $key): int|float|bool|string|null
+    {
+        return $this->variables[$key] ?? null;
+    }
+
+    /**
+     * @param string|null $environmentFile
+     *
+     * @throws \Exception
+     */
+    public function save(?string $environmentFile = null): void
```

Comment:
I think this method could return a boolean to indicate that file was written or not

Reply:
No need, if there is a problem an exception will be thrown